scipy/stats/tests/test_morestats.py

```diff
@@ -686,6 +686,70 @@ def binom_test_func(x, n=None, p=0.5, alternative='two-sided'):
 class TestBinomTest:
     """Tests for stats.binomtest."""
 
+    # Expected results here are from old binom_test function.
+    # The alternative will be left unspecified which defaults
+    # it to two-sided.
+    @pytest.mark.xfail_on_32bit("The large inputs make these tests "
+                                "sensitive to machine epsilon level")
```

Comment:
if these tests are still failing on the 32 bit platforms could we lower the values so they aren't so sensitive to the machine epsilon? 

Alternatively could we use a skipif pytest annotation that checks for 32 vs 64 bit via a sys call? 
(Skipping the test on 64 bit)

Reply:
The whole point is to check for large inputs. The existing function takes very long for this scale. Not sure how to use skipif. Any sample? Also, you meant skip on 32 bit, right?

Reply:
There are multiple ways to check for this, take a look at the note here: https://docs.python.org/3/library/platform.html#platform.architecture
as one suggestion for a way to check. 

>Not sure how to use skipif. Any sample? 

Take a look at this one using skips on factorial function from `scipy.special`, 
https://github.com/scipy/scipy/blob/277dbab1612ae32f02038a6a0df7e061c927a5cc/scipy/special/tests/test_basic.py#L1831-L1840

>Also, you meant skip on 32 bit, right?

Yes, IIUC the tests should pass on 64 bit systems, correct?

Reply:
Ok. The current "@pytest.mark.xfail_on_32bit" does cause the tests to pass (since they're ignored on 32-bit). So, everything is working now. Do you think I should switch to @pytest.mark.skipif instead? And if so, would the decorator be something like: @pytest.mark.skipif(sys.maxsize > 2**32)?

Reply:
Will it use so much physical memory that you're concerned that it may occasionally segfault/error out on 32-bit systems (i.e., even letting it run and be a known failure could still lead to issues on occasion?).

Reply:
No, its just that 32 bit machines don't have enough precision for those kinds of numbers. So, the tests always fail since they default to a p-value of 0 or something, lacking the numerical precision.

Reply:
BTW, these kinds of large inputs take a while for the current version to run (even on 64 bit machines) while with the new code you get the same result almost instantly.